xcm/src/v3/multiasset.rs

```diff
@@ -0,0 +1,871 @@
+		&self.0
+	}
+
+	/// Return the number of distinct asset instances contained.
+	pub fn len(&self) -> usize {
+		self.0.len()
+	}
+
+	/// Prepend a `MultiLocation` to any concrete asset items, giving it a new root location.
+	pub fn prepend_with(&mut self, prefix: &MultiLocation) -> Result<(), ()> {
+		self.0.iter_mut().try_for_each(|i| i.prepend_with(prefix))
+	}
+
+	/// Mutate the location of the asset identifier if concrete, giving it the same location
+	/// relative to a `target` context. The local context is provided as `context`.
+	pub fn reanchor(
+		&mut self,
+		target: &MultiLocation,
+		context: InteriorMultiLocation,
+	) -> Result<(), ()> {
+		self.0.iter_mut().try_for_each(|i| i.reanchor(target, context))
+	}
+
+	/// Return a reference to an item at a specific index or `None` if it doesn't exist.
+	pub fn get(&self, index: usize) -> Option<&MultiAsset> {
+		self.0.get(index)
+	}
+}
+
+/// A wildcard representing a set of assets.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Debug, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub enum WildMultiAsset {
+	/// All assets in Holding.
+	All,
+	/// All assets in Holding of a given fungibility and ID.
+	AllOf { id: AssetId, fun: WildFungibility },
+	/// All assets in Holding, up to `u32` individual assets (different instances of non-fungibles
+	/// are separate assets).
+	AllCounted(#[codec(compact)] u32),
+	/// All assets in Holding of a given fungibility and ID up to `count` individual assets
+	/// (different instances of non-fungibles are separate assets).
+	AllOfCounted {
+		id: AssetId,
+		fun: WildFungibility,
+		#[codec(compact)]
+		count: u32,
+	},
+}
+
+impl TryFrom<OldWildMultiAsset> for WildMultiAsset {
+	type Error = ();
+	fn try_from(old: OldWildMultiAsset) -> Result<WildMultiAsset, ()> {
+		use OldWildMultiAsset::*;
+		Ok(match old {
+			AllOf { id, fun } => Self::AllOf { id: id.try_into()?, fun: fun.try_into()? },
+			All => Self::All,
+		})
+	}
+}
+
+impl TryFrom<(OldWildMultiAsset, u32)> for WildMultiAsset {
+	type Error = ();
+	fn try_from(old: (OldWildMultiAsset, u32)) -> Result<WildMultiAsset, ()> {
+		use OldWildMultiAsset::*;
+		let count = old.1;
+		Ok(match old.0 {
+			AllOf { id, fun } =>
+				Self::AllOfCounted { id: id.try_into()?, fun: fun.try_into()?, count },
+			All => Self::AllCounted(count),
+		})
+	}
+}
+
+impl WildMultiAsset {
+	/// Returns true if `self` is a super-set of the given `inner`.
+	///
+	/// Typically, any wildcard is never contained in anything else, and a wildcard can contain any other non-wildcard.
+	/// For more details, see the implementation and tests.
+	pub fn contains(&self, inner: &MultiAsset) -> bool {
+		use WildMultiAsset::*;
+		match self {
+			AllOfCounted { count: 0, .. } | AllCounted(0) => false,
+			AllOf { fun, id } | AllOfCounted { id, fun, .. } =>
+				inner.fun.is_kind(*fun) && &inner.id == id,
+			All | AllCounted(_) => true,
```

Comment:
Do we need to compare if `inner` is `AllCounted` with a larger value?  
(I dont know enought about this stuff)

Reply:
No - `MultiAsset` is always a single asset.